app/scripts/models/auth_brokers/redirect.js

```diff
@@ -75,8 +76,11 @@ define([
       // The slight delay is to allow the functional tests time to bind
       // event handlers before the flow completes.
       var self = this;
+      var verificationRedirect = self.relier.get('verificationRedirect');
       return p().delay(100).then(function () {
-        if (self.isOriginalTab()) {
+        if (self.isOriginalTab() || self.session.oauth &&
```

Comment:
Maybe combine the 2nd half of this into a helper function `shouldFinishInVerificationTab` for clarity?